src/Symfony/Component/HttpKernel/Attribute/Serialize.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Attribute;
+
+/**
+ * Controller tag to serialize response.
+ */
+#[\Attribute(\Attribute::TARGET_METHOD)]
+class Serialize
+{
+    public function __construct(
+        public readonly int $code = 200,
+        public readonly array $headers = [],
```

Comment:
you should add phpdoc describing the structure of those arrays (`$headers` is probably `array<string, string>` for instance)

Reply:
ok, but why Psalm not violates about this?

Reply:
done

Reply:
Probably because it is not configure to require providing this info.